src/Aspire.Hosting.Yarp/YarpServiceExtensions.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Yarp;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding YARP resources to the application model.
+/// </summary>
+public static class YarpServiceExtensions
+{
+    /// <summary>
+    /// Adds a YARP container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<YarpResource> AddYarp(
+        this IDistributedApplicationBuilder builder,
+        [ResourceName] string name)
+    {
+        var resource = new YarpResource(name);
+
+        var yarpBuilder = builder.AddResource(resource)
+                      .WithHttpEndpoint(targetPort: YarpContainerImageTags.Port)
+                      .WithImage(YarpContainerImageTags.Image)
+                      .WithImageRegistry(YarpContainerImageTags.Registry)
+                      .WithEnvironment("ASPNETCORE_ENVIRONMENT", builder.Environment.EnvironmentName)
+                      .WithOtlpExporter();
+
+        if (builder.Environment.IsDevelopment())
+        {
+            // YARP will not trust the cert used by aspire otlp endpoint when running locally
+            yarpBuilder.WithEnvironment("YARP_UNSAFE_OLTP_CERT_ACCEPT_ANY_SERVER_CERTIFICATE", "true");
+        }
+
+        // Map the configuration file
+        yarpBuilder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
```

Comment:
No need to wait for any events here before passing in the the config. I know we use `AfterEndpointsAllocatedEvent` in some other resources, but it's a bit of an anti-pattern (particularly for container resources) that we should work to move away from.

Reply:
I know in this PR we only support static config files, but I want to be able to support dynamic config built by Aspire at some point. Which event should I be subscribing for that? Ideally the config should be generated just at the `Build()` time.

Reply:
You should be able to use the [callback version of WithContainerFiles](https://github.com/dotnet/aspire/blob/b72eb4fdab04b104f9ac6a09a49f8efac13b99f1/src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs#L817) as the callback is executed when the resource is being built.